### PR TITLE
DB-8559 do not include PK column in the setcolumnlist of update state…(2.7)

### PR DESCRIPTION
**Pull request by yxia92**
…ment for merge_data_from_file

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

File: splice_machine/src/test/java/com/splicemachine/derby/impl/load/UpportIT.java [hunk 1]
```diff
@@ -506,8 +506,8 @@ public void testMergeIntoOccupiedPartiallyTableWillUpdate() throws Exception {
 
         ResultSet resultSet = statement.executeQuery();
         //make sure that the bad records list is good
-        //all 5 rows are updated, no extra row inserted
-        validateMergeResults(resultSet,5,0, 0);
+        //no row should be updated, as we only use the pk column for the merge_data_from_file call
+        validateMergeResults(resultSet,0,0, 0);
```
Comment (jyuanca) [inline splice_machine/src/test/java/com/splicemachine/derby/impl/load/UpportIT.java]: do you mean non-PK columns?

Comment (yxia92) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/load/UpportIT.java]: I meant PK columns. For this test case, all the columns(essentially 'a') passed to merge_data_from_file are PK columns, so after removing the pk columns from the set column list, nothing is left, and no update statement needs to be triggered, and we should see no row updated. That is what I meant. Thanks.

Comment (jyuanca) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/load/UpportIT.java]: got it thanks!

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**